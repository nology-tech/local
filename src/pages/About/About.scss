@use "../../../src/styles/variables" as *;

.about {
  padding: 1rem;
  height: 100%;
}
.about__container {
  background-color: $white;
  border-radius: 2rem;
  border: solid $black 0.125rem;
  padding: 1rem;
  height: 100%;

  .about__heading {
    font-weight: 700;
    font-size: 1rem;
  }
}
