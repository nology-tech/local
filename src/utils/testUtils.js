import { jest } from "@jest/globals";
import { render } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";
import UserContext from "../context/UserContext";
import mockUser from "../data/mockUser";

export const customRender = (
  ui,
  useRouting = true,
  contextOptions = {
    useContext: true,
    value: { user: mockUser, setUser: jest.fn() },
  }
) => {
  // wrap components in routing and context if requested
  const routingUiResult = useRouting && wrapWithRouting(ui);
  const contextUiResult =
    contextOptions.useContext &&
    wrapWithContext(routingUiResult, contextOptions.value);

  // use RTL's render function to return the test component
  return render(contextUiResult);
};

const wrapWithRouting = (ui) => {
  return <Router>{ui}</Router>;
};

const wrapWithContext = (ui, value) => {
  return <UserContext.Provider value={value}>{ui}</UserContext.Provider>;
};
