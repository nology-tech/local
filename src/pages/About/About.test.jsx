import About from "./About";
import { customRender } from "../../utils/testUtils";

it("should render About page", () => {
  const { container } = customRender(<About />);

  expect(container).toMatchSnapshot();
});
