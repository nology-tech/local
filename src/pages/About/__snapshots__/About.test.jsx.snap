// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render About page 1`] = `
<div>
  <div
    class="layout"
  >
    <div
      class="layout__header"
    >
      <div
        class="header"
      >
        <img
          alt=""
          src="BackButton.svg"
        />
        <div
          class="header__container"
        >
          <div
            class="header__logo-title"
          >
            <img
              alt=""
              src="AppLogo.svg"
            />
            <h1>
              LOCAL
            </h1>
          </div>
        </div>
      </div>
      <img
        alt="Menu icon"
        src="menu.svg"
      />
    </div>
    <div
      class="about"
    >
      <div
        class="about__container"
      >
        <h1
          class="about__heading"
        >
          About
        </h1>
        <p
          class="about__paragraph"
        >
          About Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industrys standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industrys standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.
        </p>
      </div>
    </div>
  </div>
</div>
`;
